### ISSUE TITLE
Gulp Preview throws error

### ISSUE THREAD

**Issue by paulsjv** [tniswong/web-build#14]
When starting gulp preview after a fresh install using `npm install` I get the following error:

`gulp test` runs fine without any issue.

`$ npm -v
3.6.0`

`$ node -v
v5.5.0`

```
$ gulp preview
[12:37:58] Requiring external module babel-core/register
[12:38:01] Using gulpfile ~/git/ls/2.0/leansheets/gulpfile.babel.js
[12:38:01] Starting 'preview'...
/Users/jay.paulson/git/ls/2.0/leansheets/node_modules/gulp-jspm/lib/Bundler.js:90
        targetPathSplit = (targetPath || "").split(' '),
                                             ^

TypeError: (targetPath || "").split is not a function
    at Bundler._parseTargetPath (/Users/jay.paulson/git/ls/2.0/leansheets/node_modules/gulp-jspm/lib/Bundler.js:90:46)
    at Bundler.bundle (/Users/jay.paulson/git/ls/2.0/leansheets/node_modules/gulp-jspm/lib/Bundler.js:275:27)
    at DestroyableTransform.<anonymous> (/Users/jay.paulson/git/ls/2.0/leansheets/node_modules/gulp-jspm/lib/GulpJspmApi.js:82:22)
    at DestroyableTransform._flush (/Users/jay.paulson/git/ls/2.0/leansheets/node_modules/gulp-jspm/lib/StreamValidator.js:52:21)
    at DestroyableTransform.<anonymous> (/Users/jay.paulson/git/ls/2.0/leansheets/node_modules/readable-stream/lib/_stream_transform.js:123:12)
    at DestroyableTransform.g (events.js:273:16)
    at emitNone (events.js:80:13)
    at DestroyableTransform.emit (events.js:179:7)
    at prefinish (/Users/jay.paulson/git/ls/2.0/leansheets/node_modules/readable-stream/lib/_stream_writable.js:501:12)
    at finishMaybe (/Users/jay.paulson/git/ls/2.0/leansheets/node_modules/readable-stream/lib/_stream_writable.js:513:7)
    at endWritable (/Users/jay.paulson/git/ls/2.0/leansheets/node_modules/readable-stream/lib/_stream_writable.js:521:3)
    at DestroyableTransform.Writable.end (/Users/jay.paulson/git/ls/2.0/leansheets/node_modules/readable-stream/lib/_stream_writable.js:486:5)
    at Readable.onend (/Users/jay.paulson/git/ls/2.0/leansheets/node_modules/readable-stream/lib/_stream_readable.js:545:10)
    at Readable.g (events.js:273:16)
    at emitNone (events.js:80:13)
    at Readable.emit (events.js:179:7)
    at endReadableNT (/Users/jay.paulson/git/ls/2.0/leansheets/node_modules/readable-stream/lib/_stream_readable.js:960:12)
    at nextTickCallbackWith2Args (node.js:474:9)
    at process._tickCallback (node.js:388:17)
Segmentation fault: 11
```

**Comment by paulsjv** [comment]
Update when running `npm install` I get the following error.  Something tells me if this doesn't load things aren't going to work.

`npm ERR! git fetch -a origin (git://github.com/tniswong/jspm-cli.git) fatal: Unable to look up github.com (port 9418) (nodename nor servname provided, or not known)`

**Comment by tniswong** [comment]
This is a merge issue on StreamCompiler.js.

Specifically the defaults for the Scripts section.

Compare yours to:

```
scripts: {

    filter: [
        paths.src.js('**/*.js'),
        paths.src.templates('**/*.html'),
        `!${paths.src.js('modules/templates')}`, // excludes templates dir itself
        `!${paths.src.js('modules/templates/**/*.js')}` // excludes templates dir contents
    ],

    handler: (opts) => (stream) => {

        // special handling for minify: true since ngAnnotate requires unminified babel output from jspm.
        let jspmOpts = _.extend({}, opts);
        delete jspmOpts.minify;

        let resultStream = stream
            .pipe(manifold([

                manifold.duct(

                    paths.src.templates('**/*.html'),

                    (stream) => stream.pipe(angularTemplateCache({
                        moduleSystem: 'ES6',
                        filename: path.relative(paths.src(), paths.src.js('modules/templates/templates.js')),
                        module: 'app.templates',
                        standalone: true
                    }))

                )

            ]))
            .pipe(jspm.buildStatic(paths.src.js(entryPoint.js), `js/${APP_NAME}.js`, jspmOpts))
            .pipe(ngAnnotate());

        if (opts.minify) {
            resultStream = resultStream.pipe(uglify());
        }

        return resultStream;

    }

},
```

It's this code that does the template caching and ngAnnotate.

The the error you are experiencing is due to a modification in the `jspm.buildStatic` api. It now reads from the gulp stream, which means you pipe along all of the javascript instead of just the entryPoint. Because of this, you now have to specify the entrypoint as the first parameter.

Let me know if you still have issues.

**Comment by paulsjv** [comment]
I am redoing the merge to see if I can fix the issue.  However, now I've run into a new issue.  When I run `gulp preview` I get a 404 when the page loads.  This is due to the coding in index.html of `modules/main/main.js`.  I had to change that to where my main.js was and it loaded fine.

However, this did not fix issue #12 that I reported.
